src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/commissioner/CommissioningFragment.java

```diff
@@ -0,0 +1,169 @@
+      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    // Inflate the layout for this fragment
+    return inflater.inflate(R.layout.commissioner_commissioning_fragment, container, false);
+  }
+
+  public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    ProgressBar progressBar = getActivity().findViewById(R.id.commissioning_progress);
+    progressBar.setMin(0);
+    progressBar.setMax(100);
+
+    deviceInfo = getArguments().getParcelable(Constants.KEY_DEVICE_INFO);
+    networkInfo = getArguments().getParcelable(Constants.KEY_NETWORK_INFO);
+
+    Data arguments =
+        new Data.Builder()
+            .putString(Constants.KEY_DEVICE_INFO, new Gson().toJson(deviceInfo))
+            .putString(Constants.KEY_DEVICE_INFO, new Gson().toJson(networkInfo))
+            .build();
+    commssionerWorkRequest =
+        new OneTimeWorkRequest.Builder(CommissionerWorker.class).setInputData(arguments).build();
+
+    WorkManager.getInstance(getActivity()).enqueue(commssionerWorkRequest);
+
+    WorkManager.getInstance(getActivity())
+        .getWorkInfoByIdLiveData(commssionerWorkRequest.getId())
+        .observe(getViewLifecycleOwner(), this);
+
+    view.findViewById(R.id.cancel_button)
+        .setOnClickListener(
+            new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                WorkManager.getInstance(getActivity())
+                    .cancelWorkById(commssionerWorkRequest.getId());
+
+                CommissionerActivity commissionerActivity = (CommissionerActivity) getActivity();
+                commissionerActivity.finishCommissioning(Activity.RESULT_CANCELED);
+              }
+            });
+
+    view.findViewById(R.id.done_button)
+        .setOnClickListener(
+            new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                CommissionerActivity commissionerActivity = (CommissionerActivity) getActivity();
+                commissionerActivity.finishCommissioning(Activity.RESULT_OK);
+              }
+            });
+  }
+
+  @Override
+  public void onChanged(@Nullable WorkInfo workInfo) {
+    TextView statusText = getActivity().findViewById(R.id.status_text);
+    ProgressBar progressBar = getActivity().findViewById(R.id.commissioning_progress);
+
+    if (workInfo != null) {
+      if (workInfo.getState().isFinished()) {
+
+        showCommissionDone(
+            workInfo.getOutputData().getBoolean(Constants.KEY_SUCCESS, false),
+            workInfo.getOutputData().getString(Constants.KEY_COMMISSIONING_STATUS));
+      } else {
+        showInProgress(workInfo.getProgress().getString(Constants.KEY_COMMISSIONING_STATUS));
+      }
+    }
+  }
+
+  private void showInProgress(String status) {
+    TextView statusText = getActivity().findViewById(R.id.status_text);
```

Comment:
Given that statusText is used in multiple methods of this file, extract it to a class variable in onViewCreated() and use the variable here and other places.
Same for other views that have been references multiple times.

Reply:
updated.